### PR TITLE
Leo/bsi b soup

### PR DESCRIPTION
**Pull request by adamjanovsky**

### CODE REVIEW

**Comment by adamjanovsky** [comment]
## As of 9 July, the following is done:

- [x] Traversal of BSI webpage, only active certificates

## As of 9 July, the following remains to be done:

- [ ] Full traversal of archived certificates
- [ ] Extraction of all information about a single certificate
- [ ] Feed the BSI outputs into `CommonCriteriaCert` class
- [ ] Output the mismatch between BSI source and CommonCriteria.org source
- [ ] Select 3 certificates and verify (using hash function) that their reports are the same

File: sec_certs/bsi_parse.py [hunk 1]
```diff
@@ -136,23 +101,23 @@ class Bsitmp(ComplexSerializableType):
...
 
-subpage = Bsitmp(test_url)
-
-
-# print(subpage.certification_date)
-
-
-def process(root_url: str):
-    browser = BsiBrowser(root_url)
+def process(base_url: str):
```
Comment (adamjanovsky) [inline sec_certs/bsi_parse.py:119]: It feels that this whole method should be part of `BsiBrowser.parse()` method.

File: sec_certs/bsi_parse.py [hunk 2]
```diff
@@ -136,23 +101,23 @@ class Bsitmp(ComplexSerializableType):
...
+    browser = BsiBrowser(base_url)
     browser.parse()
     tmp_list = []
     for handler in browser.handler_list:
```
Comment (adamjanovsky) [inline sec_certs/bsi_parse.py:123]: This whole for loop can be written as a list comprehension and you can call `tmp_list.extend([the_comprehension])`

File: sec_certs/bsi_parse.py [hunk 3]
```diff
@@ -16,115 +16,80 @@
...
+
+        # ------------------------------------------------------------------------------------------------
+
+
+
+# To end the processing, a last class will be used with final links
+# to retrieve simple data written on the page
 
 
 class Bsitmp(ComplexSerializableType):
```
Comment (adamjanovsky) [inline sec_certs/bsi_parse.py:95]: Name of class is not really explanatory. Should be named something like `BSICertificate`

File: sec_certs/bsi_parse.py [hunk 4]
```diff
@@ -161,33 +126,15 @@ def process(root_url: str):
             temp = Bsitmp(url)
             # print(temp.html_content)
             tmp_list.append(temp)
-        for link in handler.archive_list:
-            arch_handler = BsiHandler(link)
+        for arch_handler in handler.handler_list:
+            for arch_url in arch_handler.link_list:
+                arch_temp = Bsitmp(arch_url)
+                tmp_list.append(arch_temp)
```
Comment (adamjanovsky) [inline sec_certs/bsi_parse.py:132]: Same here, use comprehension and `.extend()` notation

File: sec_certs/bsi_parse.py [hunk 5]
```diff
@@ -136,23 +101,23 @@ class Bsitmp(ComplexSerializableType):
...
-
-# print(subpage.certification_date)
-
-
-def process(root_url: str):
-    browser = BsiBrowser(root_url)
+def process(base_url: str):
+    browser = BsiBrowser(base_url)
     browser.parse()
     tmp_list = []
```
Comment (adamjanovsky) [inline sec_certs/bsi_parse.py:122]: this could then be instance variable

File: sec_certs/bsi_parse.py [hunk 6]
```diff
@@ -16,115 +16,80 @@
...
 
-"""
-To end the processing, a last class will be used with final links
-to retrieve simple data written on the page
-"""
+        self.handler_list = \
+            [BsiHandler("https://www.bsi.bund.de/" + a['href'])
+             for a in self.soup.find_all('a', href=True, recursive=True, title=re.compile('Archive'))]
+
+        if len(self.handler_list) != 0:
```
Comment (adamjanovsky) [inline sec_certs/bsi_parse.py:83]: `if self.handler_list:`

**Review by adamjanovsky — COMMENTED** [review]
Some intermediate review. Once done with your work, be sure to get rid of non-descriptive comments and make the code (at least somewhat) PEP-8 compliant.

File: sec_certs/bsi_parse.py [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+class Bsitmp(ComplexSerializableType):
+    """
+    a class that will contain essentials data to compare to CC certs
+    """
+    html_content: list
+    certification_date: str
+    valid_until: str
+    soup: BeautifulSoup
+    id: str
+    pdf_links: list[str]
```
Comment (adamjanovsky) [inline sec_certs/bsi_parse.py:104]: this won't work. You need to use `List[str]`

File: sec_certs/bsi_parse.py [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+        self.link_list = []
+        self.handler_list = []
+        self.archive_list = []
+
+    def parse(self):
+
+        self.handler_list = \
+            [BsiHandler("https://www.bsi.bund.de/" + a['href'])
+             for a in self.soup.find_all('a', href=True, recursive=True)
+             if 'c-navigation-teaser' in str(a.get('class'))]
```
Comment (adamjanovsky) [inline sec_certs/bsi_parse.py:55]: You can search directly by link class with `find_all(class='c-navigation-teaser')`

**Comment by J08nY** [comment]
This is now done in #247.

**Pull request closed without merging**